server/src/main/java/com/lunchtime/implementation/RestaurantServiceImplement.java

```diff
@@ -0,0 +1,74 @@
+package com.lunchtime.implementation;
+
+import com.lunchtime.models.Restaurant;
+import com.lunchtime.repository.RestaurantRepository;
+import com.lunchtime.service.PersonService;
+import com.lunchtime.service.RestaurantService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class RestaurantServiceImplement implements RestaurantService {
+
+    private final RestaurantRepository restaurantRepository;
+    private final PersonService personService;
+
+    public RestaurantServiceImplement(RestaurantRepository restaurantRepository, PersonService personService) {
+        this.restaurantRepository = restaurantRepository;
+        this.personService = personService;
+    }
+
+    public Restaurant save(Restaurant restaurant) {
+        return personService.findById(restaurant.getPerson().getId())
+            .map(person -> {
+                restaurant.setPerson(person);
+                return restaurantRepository.save(restaurant);
+            })
+            .orElseGet(() -> {
+                return null;
+            });
+    }
+
+    public Page<Restaurant> findAll(Pageable pageable) {
+        return restaurantRepository.findAll(pageable);
+    }
+
+    public Optional<Restaurant> findById(Long id) {
+        return restaurantRepository.findById(id);
+    }
+
+    public Restaurant update(Restaurant newRestaurant) {
+        return findById(newRestaurant.getId())
+            .map(restaurant -> {
+                restaurant.setName(newRestaurant.getName());
+                restaurant.setEmail(newRestaurant.getEmail());
+                restaurant.setTextAddress(newRestaurant.getTextAddress());
+                restaurant.setWebsite(newRestaurant.getWebsite());
+                restaurant.setDescription(newRestaurant.getDescription());
+                restaurant.setWorkingTime(newRestaurant.getWorkingTime());
+                restaurant.setMenuId(newRestaurant.getMenuId());
+                restaurant.setPerson(newRestaurant.getPerson());
+                restaurant.setTables(newRestaurant.getTables());
+                restaurant.setLongitude(newRestaurant.getLongitude());
+                restaurant.setLatitude(newRestaurant.getLatitude());
+                return save(restaurant);
+            })
+            .orElseGet(() -> {
+                return null;
+            });
+    }
+
+    public Restaurant delete(Long id) {
+        return findById(id)
+            .map(restaurant -> {
+                restaurant.setIsDeleted(true);
+                return save(restaurant);
```

Comment:
I think we should use `setisDeleted()` method with no args for safety. Also I'd rename it to `setDeleted()` and create another method `setAvailable()` or `setVisible()` or even `setNotDeleted()`, where we can "restore" entity from deleted ones if needed.

Also take care about `.map()` method. This code looks very cool with `.map()`, but it refers more to multiple elements and/or collections.

Reply:
Or even better. We can use `setInactive()` and `setActive()` methods. In fact, we don't delete entities, but mark them as non-shown.

Reply:
I agree that we can change `setDelete()` method signature. But it must be delete, because user push button delete and he or she expects that entity will be deleted. Also there is common practice in this property naming.
`setInActive() setActive()` can be applied to `Person` who doesn't confirm email

Reply:
@narekacy 
I cant find information that using `.map()` in this case is a bad practice. Optional interface provide this method and repository can use it.
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html
https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#findById-ID-
Could you explain why it is bad practice?

Reply:
As my task is "Show personal info" I created new task in trello to fix this issue https://trello.com/c/HB5kLggh/95-90-change-setdelete-method-signature

Reply:
> Also take care about `.map()` method. This code looks very cool with `.map()`, but it refers more to multiple elements and/or collections.

No no, I don't call it bad practiced, no :) I just wanted to express my opinion that `.map()` in Java is more about Stream API for collections, that's it :)

This code looks great and I like it for sure, a fortiory, it works!

Reply:
Thanks for explanation. I'm newbie in Steam API and I didn't find any mentions about `stream` in `java.util.Optional` class